app/backend/src/app.py

```diff
@@ -97,51 +97,32 @@ def log_unhandled_exception(exc_type, exc_value, exc_traceback):
 logger.info(f"Starting")
 
 if config.config["ROLE"] in ["api", "all"]:
-    auth = Auth()
-    open_server = grpc.server(
-        futures.ThreadPoolExecutor(8), interceptors=[ErrorSanitizationInterceptor(), TracingInterceptor()]
-    )
-    open_server.add_insecure_port("[::]:1752")
-    auth_pb2_grpc.add_AuthServicer_to_server(auth, open_server)
-    bugs_pb2_grpc.add_BugsServicer_to_server(Bugs(), open_server)
-    resources_pb2_grpc.add_ResourcesServicer_to_server(Resources(), open_server)
-    open_server.start()
-
-    jailed_server = grpc.server(
-        futures.ThreadPoolExecutor(8),
-        interceptors=[
-            ErrorSanitizationInterceptor(),
-            TracingInterceptor(),
-            auth.get_auth_interceptor(allow_jailed=True),
-        ],
-    )
-    jailed_server.add_insecure_port("[::]:1754")
-    jail_pb2_grpc.add_JailServicer_to_server(Jail(), jailed_server)
-    jailed_server.start()
-
-    servicer = API()
     server = grpc.server(
         futures.ThreadPoolExecutor(64),
         interceptors=[
             ErrorSanitizationInterceptor(),
             TracingInterceptor(),
-            auth.get_auth_interceptor(allow_jailed=False),
+            AuthValidatorInterceptor(),
         ],
     )
     server.add_insecure_port("[::]:1751")
 
     account_pb2_grpc.add_AccountServicer_to_server(Account(), server)
-    api_pb2_grpc.add_APIServicer_to_server(servicer, server)
+    api_pb2_grpc.add_APIServicer_to_server(API(), server)
+    auth_pb2_grpc.add_AuthServicer_to_server(Auth(), open_server)
```

Comment:
I am using this PR as a base, and my code crashes on `open_server` not being defined, while being used here.